Modules/posixmodule.c

```diff
@@ -3007,6 +3015,13 @@ os_chmod_impl(PyObject *module, path_t *path, int mode, int dir_fd,
         return NULL;
 #endif
 
+    if (PySys_Audit("os.chmod", "OiO",
+                    path->object ? path->object : Py_None, mode,
+                    dir_fd == DEFAULT_DIR_FD ? Py_None : PyLong_FromLong(dir_fd)
```

Comment:
We may just need to pick an "obvious" value to pass rather than None here... it's a shame that DEFAULT_DIR_FD is so arbitrary.

I'd rather not have to manage an object's lifetime around the PySys_Audit call. Passing `0` or `-1` (assuming those aren't valid FDs? Are they?) would be obvious enough to me.

Reply:
According to documentation from [Linux](http://man7.org/linux/man-pages/man2/open.2.html), [FreeBSD](https://www.freebsd.org/cgi/man.cgi?query=open&sektion=2), [macOS](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/open.2.html) and [Microsoft C Run-Time Library](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/open-osfhandle?view=vs-2019), file descriptors are non-negative integers, negative values are used to indicate errors, so here we can safely use `-1` to indicate that `dir_fd` is not specified.